src/util/mw-api.ts

```diff
@@ -72,35 +72,40 @@ export async function getArticlesByIds(_articleIds: string[], downloader: Downlo
     );
 }
 
-export async function getArticlesByNS(ns: number, downloader: Downloader, _gapContinue?: string, continueLimit?: number): Promise<void> {
-
-    const { articleDetails: _articleDetails, gapContinue } = await downloader.getArticleDetailsNS(ns, _gapContinue);
-
-    const articleDetails = mwRetToArticleDetail(downloader, _articleDetails);
-
-    const numDetails = Object.keys(articleDetails).length;
-    await articleDetailXId.setMany(articleDetails);
-
-    for (const [articleId, articleDetail] of Object.entries(_articleDetails)) {
-        await redirectsXId.setMany(
-            (articleDetail.redirects || []).reduce((acc, redirect) => {
-                const rId = redirect.title.replace(/ /g, '_');
-                return {
-                    ...acc,
-                    [rId]: { targetId: articleId, title: redirect.title },
-                };
-            }, {}),
-        );
-    }
+export async function getArticlesByNS(ns: number, downloader: Downloader, continueLimit?: number): Promise<void> {
+    let _gapContinue: string;
+    do {
+        const { articleDetails: _articleDetails, gapContinue } = await downloader.getArticleDetailsNS(ns, _gapContinue);
+        _gapContinue = gapContinue;
+        const articleDetails = mwRetToArticleDetail(downloader, _articleDetails);
+
+        const numDetails = Object.keys(articleDetails).length;
+        await articleDetailXId.setMany(articleDetails);
+
+        for (const [articleId, articleDetail] of Object.entries(_articleDetails)) {
+            await redirectsXId.setMany(
+                (articleDetail.redirects || []).reduce((acc, redirect) => {
+                    const rId = redirect.title.replace(/ /g, '_');
+                    return {
+                        ...acc,
+                        [rId]: { targetId: articleId, title: redirect.title },
+                    };
+                }, {}),
+            );
+        }
 
-    logger.log(`Got [${numDetails}] articles from namespace [${ns}]`);
-    totalArticles += numDetails;
-    const canContinue = typeof continueLimit === 'undefined' || continueLimit > 0; // used for testing
+        logger.log(`Got [${numDetails}] articles from namespace [${ns}]`);
+        totalArticles += numDetails;
 
-    if (gapContinue && canContinue) {
-        const nextContinueLimit = typeof continueLimit === 'undefined' ? undefined : continueLimit - 1;
-        return getArticlesByNS(ns, downloader, gapContinue, nextContinueLimit);
-    }
+        // Only for testing purposes
+        if (typeof continueLimit !== 'undefined') {
```

Comment:
My personal opinion is that having some code only for testing purposes (outside the tests) would be considered as anti-pattern. However, I'd not mind about this in this particular case and I don't see another reasonable way to limit the number of iterations here.
Anyways, that would be shortened to the following one-liner:
```
if (--(continueLimit as number) < 0) break;
```
There's a [sandbox](https://www.typescriptlang.org/play/?module=1&ssl=25&ssc=52&pln=25&pc=9#code/DYUwLgBAdhC8EEYAMBuAsAKFJAlnCqmA9EdhAMYD2UYOUAriADI4C2OYAXNPawEYgATvgCs6DCTJUadRi3aR49KABMQAMzogV4zJnVQAFCICUujOuXla1CAcPTaDZmw4B+bg35CTEAN6YEEEQKpT+gcGR0gDOlKAAdMCUAOaGAAYcQhAAJH44AL5pZhGRQTgA1OXmRABUJaU46hCGYACeAA4glE2Osi4KEACEsPAA5MpqmlDao74BGKWLEI3Nvc7yHBAAfARz9Uula3KuihTUTscDALSI4geR+RAgwNEg4Qv3izFxIIkphqMQAAPDizO6fUp8QQgACGAGtwZ98vtgsiMDUiHoPosVoYrlcHOc+htIDDojxvIJfAAeXYQKGwhEooL1R4AdwAFjhQM08LSoMUMPVvgkkqlRqFpmDMPkgA) where I've tested this.

Reply:
Agree with that comment

Reply:
Solved.